### DISCUSSION TITLE
WiFi Module for Mspa Whirlpool?

### DISCUSSION THREAD

**Discussion by Zenyte-zz** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#620] [Q&A]
Hi Community, i like this project 
is there any chance to adapt this project to my Whirlpool mSpa "Frame" 
The Motherboard of this Spa looks like this https://www.tubs.parts/en/mspa-urbanmuseframe-2021-pcb-p-5609.html

**Comment by visualapproach** [comment]
Sorry, I have no clue

**Reply by Zenyte-zz** [reply]
Hi visualapproach, thank you for reply and for your work on this project. 
Maybe someone from the Community will read this and give me a hint or have done this on a mSPA.

**Reply by visualapproach** [reply]
You are welcome! Maybe better chance if you post some pictures of your pump's guts.

**Reply by johannesc** [reply]
A bit late, but I have recently added (so far basic) Wi-Fi support to my M-Spa Aurora Urban. My control board looks like the one on your picture. See https://github.com/johannesc/mspa-wifi for more information.

**Comment by phrozendevil** [comment]
what a great project! i would be very happy to have this working on my pool.
i have a MSpa pool too, to be exact [this one](https://www.miweba.de/whirlpool/outdoor-whirlpool/23417/miweba-mspa-whirlpool-comfort-ottoman-c-om061-whirlpool-aufblasbar-in-rattan-optik-6-personen)

the pump is fully integrated in the pool and there is a cable remote with all the buttons and display, the controller in there is a PIC16F723A which is communicating with the mainboard (also a PIC16F723A) via a 4-pin cable (it has +5V, RX, TX, GND).

[this is the mainboard](https://www.miweba.de/ersatzteile/miweba-mspa-whirlpools/mspa-modellreihe-20212022/mspa-comfort-bergen-c-be041/22707/mspa-mainboard-b9301486-lite/comfort-serie-2021-2022?number=PR0026386-01) or [better pic](https://www.tubs.parts/de/mspa-hauptplatine-litecomfort-2021-p-5774.html)
[this is the remote](https://www.boospa.net/en/ss21-carte-electronique-telecommande-avec-uv-comfort.html)

i almost did a lot of research, but not sure if it would work with your module. do you think there is a chance to make it work with it?

many thanks in advance
Thorsten

**Reply by visualapproach** [reply]
Thanks! I don't know. If I had to bet, I'd say no.

**Reply by schnaidi** [reply]
Hey hey I wanted to ask if there is any news because I also have an mspa pool myself and unfortunately still haven't found the right way for it

**Reply by johannesc** [reply]
Depending on model my project might work, see https://github.com/johannesc/mspa-wifi.

**Reply by phrozendevil** [reply]
> Depending on model my project might work, see https://github.com/johannesc/mspa-wifi.

WOW great, will build it as soon as possible. Thanks a lot!!

**Reply by Speedafix** [reply]
Also ich habe es mir einem wemos demon Mini d1 hinbekommen. Alte Fernbedienung ab und diesen dran. Mein Code läuft sauber und erhält Strom.
Es muss dem Grund nach bei jedem Modell gehen und die Switch müssen nur erweitert werden. Ich nutze den mspa tekapo nun ausschließlich über Homeassistant
